src/envoy/http/path_matcher/filter_test.cc

```diff
@@ -170,6 +170,23 @@ TEST_F(PathMatcherFilterTest, DecodeHeadersNoMatch) {
                     ->value());
 }
 
+TEST_F(PathMatcherFilterTest, DecodeHeadersMissingHeaders) {
+  Envoy::Http::TestRequestHeaderMapImpl missingPath{{":method", "POST"}};
+  Envoy::Http::TestRequestHeaderMapImpl missingMethod{{":path", "/bar"}};
+
+  // Filter should reject this request
+  EXPECT_CALL(
+      mock_cb_.stream_info_,
+      setResponseFlag(
+          Envoy::StreamInfo::ResponseFlag::UnauthorizedExternalService))
```

Comment:
Why it is UnauthorizedExternalService? I saw it is rejected by 400 so where it is set?

Reply:
In `rejectReject` sets this, it's independent of the HTTP status code. None of the other enum values seem like a better match.

Reply:
Oh, I see.